pubsub/kubemq/metadata.go

```diff
@@ -0,0 +1,85 @@
+package kubemq
+
+import (
+	"fmt"
+	"github.com/dapr/components-contrib/pubsub"
+	"strconv"
+	"strings"
+)
+
+type metadata struct {
+	host              string
+	port              int
+	clientID          string
+	authToken         string
+	group             string
+	isStore           bool
+	useMock           bool
+	disableReDelivery bool
+}
+
+func parseAddress(address string) (string, int, error) {
+	var host string
+	var port int
+	var err error
+	hostPort := strings.Split(address, ":")
+	if len(hostPort) != 2 {
+		return "", 0, fmt.Errorf("invalid kubeMQ address, address format is invalid")
+	}
+	host = hostPort[0]
+	if len(host) == 0 {
+		return "", 0, fmt.Errorf("invalid kubeMQ address, host is empty")
+	}
+	port, err = strconv.Atoi(hostPort[1])
+	if err != nil {
+		return "", 0, fmt.Errorf("invalid kubeMQ address, port is invalid")
+	}
+	return host, port, nil
+}
+
+// createMetadata creates a new instance from the pubsub metadata
+func createMetadata(pubSubMetadata pubsub.Metadata) (*metadata, error) {
+	result := &metadata{}
+	if val, found := pubSubMetadata.Properties["address"]; found && val != "" {
+		var err error
+		result.host, result.port, err = parseAddress(val)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		return nil, fmt.Errorf("invalid kubeMQ address, address is empty")
+	}
+	if val, found := pubSubMetadata.Properties["clientID"]; found && val != "" {
+		result.clientID = val
+	}
+
+	if val, found := pubSubMetadata.Properties["authToken"]; found && val != "" {
```

Comment:
Other Dapr components read the Kubernetes auth token from the volume that is mounted by Kubernetes. I would recommend doing that rather than requiring an auth token passed as metadata - unless KubeMQ works differently

Reply:
This is the client side. if you want to provide a auth token, you should provide this.
it works little bit different.

Reply:
Ok so this is not the same token that is used to access the K8s APIs?

Reply:
No, it's not. this is auth token to access KubeMQ server